templates_test.go

```diff
@@ -377,3 +380,55 @@ func TestTemplateModeRemote(t *testing.T) {
 }
 
 // TODO: test typed errors for custom and remote (embedded checked)
+
+func TestTemplateFuncYAML(t *testing.T) {
+	// create test directory
+	root := "testdata/testTemplateEmbeddedYAML"
+	defer using(t, root)()
+
+	// Client whose internal templates will be used.
+	client := fn.New(
+		fn.WithRegistry(TestRegistry),
+		fn.WithRepositories("testdata/repositories"))
+
+	// write out a template
+	err := client.Create(fn.Function{
+		Root:     root,
+		Runtime:  "manifestedRuntime",
+		Template: "customLanguagePackRepo/customTemplate",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Assert file exists as expected
+	_, err = os.Stat(filepath.Join(root, "func.yaml"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	funcByteArray, err := ioutil.ReadFile(filepath.Join(root, "func.yaml"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	config := fn.Config{}
+	err = yaml.Unmarshal(funcByteArray, &config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testVariableName := "TEST_VARIABLE"
+	testVariableValue := "test"
```

Comment:
This is good, but this is really testing a side-effect.  The intent is to ensure that BuildEnvs set in the Template are affected on the resultant Function, so you could actually save a little work, and test more of the process overall, by perhaps creating a Function from the Template and validating it looks as expected:
```go
f, err := fn.NewFunction(root)
if err != nil {
  t.Fatal(err)
}
if !reflect.DeepEqual(f.BuildEnvs, envs) {
```
.. and then testing if f.BuildEnvs is